src/menu/menu.c

```diff
@@ -1021,6 +1021,79 @@ update_client_list_combined_menu(struct server *server)
 	menu_create_scene(menu);
 }
 
+static bool
+is_ixoth(char *filename)
+{
+	struct stat sb;
+	return stat(filename, &sb) == 0 && (sb.st_mode & S_IXOTH);
+}
+
+static bool
+isprog(const char *filename, gchar **paths)
+{
+	/* handle filename with absolute path */
+	if (filename[0] == '/') {
+		return is_ixoth((char *)filename);
+	}
+
+	/* handle filename with relative path by iterating over $PATH */
+	for (gchar **p = paths; *p; p++) {
+		char path[4096];
+		snprintf(path, sizeof(path), "%s/%s", *p, filename);
+		if (is_ixoth(path)) {
+			return true;
+		}
+	}
+	return false;
+}
+
+static void
+try_to_add_terminal(struct menu *menu)
+{
+	const char * const terminals[] = {
+		"alacritty",
```

Comment:
Should this list start with `x-terminal-emulator`? That should respect the user configuration on Debian based systems and be simply ignored on other distros.